File: exercise/week04/c#/Construct/ConstructInventory.cs
namespace Construct;

public class ConstructInventory(Artefact[] artefacts)
{
    public void UpdateSimulation()
    {
        for (int i = 0; i < artefacts.Length; i++)
        {
            var artefact = artefacts[i];

            if (artefact.Name != "Aged Signal" &&
                artefact.Name != "Backdoor Pass to TAFKAL80ETC Protocol")
            {
                if (artefact.Integrity > 0)
                {
                    if (artefact.Name != "Sulfuras Core Fragment")
                    {
                        artefact.Integrity--;
                    }
                }
            }
            else
            {
                if (artefact.Integrity < 50)
                {
                    artefact.Integrity++;

                    if (artefact.Name == "Backdoor Pass to TAFKAL80ETC Protocol")
                    {
                        if (artefact.TimeToLive < 11 && artefact.Integrity < 50)
                        {
                            artefact.Integrity++;
                        }

                        if (artefact.TimeToLive < 6 && artefact.Integrity < 50)
                        {
                            artefact.Integrity++;
                        }
                    }
                }
            }

            if (artefact.Name != "Sulfuras Core Fragment")
            {
                artefact.TimeToLive--;
            }

            if (artefact.TimeToLive < 0)
            {
                if (artefact.Name != "Aged Signal")
                {
                    if (artefact.Name != "Backdoor Pass to TAFKAL80ETC Protocol")
                    {
                        if (artefact.Integrity > 0)
                        {
                            if (artefact.Name != "Sulfuras Core Fragment")
                            {
                                artefact.Integrity--;
                            }
                        }
                    }
                    else
                    {
                        artefact.Integrity = 0;
                    }
                }
                else
                {
                    if (artefact.Integrity < 50)
                    {
                        artefact.Integrity++;
                    }
                }
            }
        }
    }
}

File: exercise/week04/php/src/Construct/ConstructInventory.php
<?php

namespace Construct;

class ConstructInventory
{
    /** @var Artefact[] */
    private array $artefacts;

    public function __construct(array $artefacts)
    {
        $this->artefacts = $artefacts;
    }

    public function updateSimulation(): void
    {
        foreach ($this->artefacts as $artefact) {
            if ($artefact->name !== "Aged Signal" &&
                $artefact->name !== "Backdoor Pass to TAFKAL80ETC Protocol") {

                if ($artefact->integrity > 0 && $artefact->name !== "Sulfuras Core Fragment") {
                    $artefact->integrity--;
                }

            } else {
                if ($artefact->integrity < 50) {
                    $artefact->integrity++;

                    if ($artefact->name === "Backdoor Pass to TAFKAL80ETC Protocol") {
                        if ($artefact->timeToLive < 11 && $artefact->integrity < 50) {
                            $artefact->integrity++;
                        }
                        if ($artefact->timeToLive < 6 && $artefact->integrity < 50) {
                            $artefact->integrity++;
                        }
                    }
                }
            }

            if ($artefact->name !== "Sulfuras Core Fragment") {
                $artefact->timeToLive--;
            }

            if ($artefact->timeToLive < 0) {
                if ($artefact->name !== "Aged Signal") {
                    if ($artefact->name !== "Backdoor Pass to TAFKAL80ETC Protocol") {
                        if ($artefact->integrity > 0 && $artefact->name !== "Sulfuras Core Fragment") {
                            $artefact->integrity--;
                        }
                    } else {
                        $artefact->integrity = 0;
                    }
                } elseif ($artefact->integrity < 50) {
                    $artefact->integrity++;
                }
            }
        }
    }

    public function getArtefacts(): array
    {
        return $this->artefacts;
    }
}
